### PR TITLE
docs: create vitepress site

### PR DESCRIPTION
**Pull request by Lehoczky**
Resolves https://github.com/vue-a11y/eslint-plugin-vuejs-accessibility/issues/809

What has been done:
- initialized vitepress with basic settings
- enabled local search functionality
- current version is read from the `package.json` and show in the navbar
- created a table component which shows every rule and whether they are in the recommended config by reading `src/configs/recommended.ts`
- markdown files in the `docs/rules` folder are automatically added to the sidebar with `getRulesForSideBar()` function (vitepress can't do this automatically, see https://github.com/vuejs/vitepress/issues/1737)
- reorganized rule specific docs:
  - resources are added to the end of the files like in https://eslint.vuejs.org/, https://typescript-eslint.io/, https://eslint.org/
  - added unicode icons to each heading
  - renamed "Rule Details" to "Options
  - wrapped `vue` code blocks with `<template>` tag, because otherwise the syntax highlighting can break in vscode and on the site

TODO:
- because I don't know where the site will be deployed, I added a `TODO` comment where we need to add the URL later 
- github action to automatically deploy the site when there is a change in the `docs` folder or in `src/configs/recommended.ts`

Screens:

image
image
image
image

### CODE REVIEW

File: docs/.vitepress/config.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
+import { defineConfig } from "vitepress";
+import { rules } from "./rulesForSidebar";
+import { version } from "../../package.json";
+
+// https://vitepress.dev/reference/site-config
+export default defineConfig({
+  title: "vuejs-accessibility",
```
Comment (Lehoczky) [inline docs/.vitepress/config.ts:7]: "eslint-plugin-vuejs-accessibility" was too long, it overflowed from the sidebar:

image

**Comment by vhoyer** [comment]
maybe we can use `https://vue-a11y.github.io/eslint-plugin-vuejs-accessibility/`, as in using Github Pages

**Comment by vhoyer** [comment]
other than that, great work, and sorry for the late reply, had a lot on my plate recently

**Comment by vhoyer** [comment]
if you can configure the github action to deploy to the `docs` branch, it would be great, if not, just let me know

**Comment by Lehoczky** [comment]
> other than that, great work, and sorry for the late reply, had a lot on my plate recently

Aww Thank you, and no worries<3

**Comment by Lehoczky** [comment]
I didn't mean to close this, just synced my fork with the main branch and this got closed automatically 😅 I'm gonna create a new PR

**Comment by Lehoczky** [comment]
> if you can configure the github action to deploy to the `docs` branch, it would be great, if not, just let me know

Done 👍

**Pull request closed without merging**